### PR TITLE
fix(Core/Spells): Flasks should not be overriden by elixirs.

### LINKED ISSUE #10129: Elixirs override flasks
### What client do you play on?

enUS

### Faction

- [X] Alliance
- [X] Horde

### Content Phase:

- [X] Generic
- [ ] 1-19
- [ ] 20-29
- [ ] 30-39
- [ ] 40-49
- [ ] 50-59
- [ ] 60

### Current Behaviour

Original report: https://github.com/chromiecraft/chromiecraft/issues/2765

If you are using an elixir while a flask is active, the flask is replaced by the elixir.

Tester:

The issue is confirmed. Both types of elixirs, guardian and battle, replace the effect of flasks on use.

### Expected Blizzlike Behaviour

Having a flask active should prevent the player from using a battle or guardian elixir. This should also affect Jujus.

### Source

> Subsequent flasks imbibed will overwrite the effects of the former and prevent a player from using other elixirs.
(source: https://legacy.zam.com/wiki/Flasks_%28WoW%29)

### Steps to reproduce the problem

Original report:
1. Use a flask
2. Use an elixir

I tested with [Flask of Supreme Power] and [Arcane Elixir].

Tester:
1. `.additem 13512`
2. `.additem 13447`
3. Use [Flask of Supreme Power]
4. Use [Elixir of Sages] (guardian elixir)
5. `.additem 13512`
6. `.additem 12451`
7. `.additem 12384`
8. Use [Flask of Supreme Power]
9. Use [Juju Power] (battle elixir)

### Extra Notes

_No response_

### AC rev. hash/commit

https://github.com/chromiecraft/azerothcore-wotlk/commit/c1ebda66bae939e7151fc9f43176dd897e8c01b7

### Operating system

Ubuntu

### Modules

- [mod-ah-bot](https://github.com/azerothcore/mod-ah-bot)
- [mod-cfbg](https://github.com/azerothcore/mod-cfbg)
- [mod-chat-transmitter](https://github.com/azerothcore/mod-chat-transmitter)
- [mod-chromie-xp](https://github.com/azerothcore/mod-chromie-xp)
- [mod-desertion-warnings](https://github.com/azerothcore/mod-desertion-warnings)
- [mod-duel-reset](https://github.com/azerothcore/mod-duel-reset)
- [mod-eluna-lua-engine](https://github.com/azerothcore/mod-eluna-lua-engine)
- [mod-ip-tracker](https://github.com/azerothcore/mod-ip-tracker)
- [mod-low-level-arena](https://github.com/azerothcore/mod-low-level-arena)
- [mod-multi-client-check](https://github.com/azerothcore/mod-multi-client-check)
- [mod-pvp-titles](https://github.com/azerothcore/mod-pvp-titles)
- [mod-pvpstats-announcer](https://github.com/azerothcore/mod-pvpstats-announcer)
- [mod-queue-list-cache](https://github.com/azerothcore/mod-queue-list-cache)
- [mod-server-auto-shutdown](https://github.com/azerothcore/mod-server-auto-shutdown)
- [lua-carbon-copy](https://github.com/55Honey/Acore_CarbonCopy)
- [lua-exchange-npc](https://github.com/55Honey/Acore_ExchangeNpc)
- [lua-custom-worldboss](https://github.com/55Honey/Acore_CustomWorldboss)
- [lua-level-up-reward](https://github.com/55Honey/Acore_LevelUpReward)
- [lua-recruit-a-friend](https://github.com/55Honey/Acore_RecruitAFriend)
- [lua-send-and-bind](https://github.com/55Honey/Acore_SendAndBind)
- [lua-temp-announcements](https://github.com/55Honey/Acore_TempAnnouncements)
- [lua-zonecheck](https://github.com/55Honey/acore_Zonecheck)
- [lua-zone-debuff](https://github.com/55Honey/Acore_ZoneDebuff)

### Customizations

None

### Server

ChromieCraft

### PR DESCRIPTION
**Pull request by UltraNix**
Fixed #10129

## Issues Addressed:

- Closes #10129
- Closes https://github.com/chromiecraft/chromiecraft/issues/2765

## Tests Performed:

- Tested ingame.

## How to Test the Changes:

Original report:

Use a flask
Use an elixir
I tested with [Flask of Supreme Power] and [Arcane Elixir].

Tester:

.additem 13512
.additem 13447
Use [Flask of Supreme Power]
Use [Elixir of Sages] (guardian elixir)
.additem 13512
.additem 12451
.additem 12384
Use [Flask of Supreme Power]
Use [Juju Power] (battle elixir)

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

File: src/server/game/Spells/SpellInfo.cpp [hunk 1]
```diff
@@ -2837,3 +2837,50 @@ void SpellInfo::_UnloadImplicitTargetConditionLists()
...
+bool SpellInfo::CheckElixirStacking(Unit const* caster) const
+{
+    if (!caster)
+    {
+        return true;
+    }
+
+    // xinef: check spell group
+    uint32 groupId = sSpellMgr->GetSpellGroup(Id);
+    if (groupId != SPELl_GROUP_GUARDIAN_AND_BATTLE_ELIXIRS)
```
Comment (claudiodfc) [inline src/server/game/Spells/SpellInfo.cpp:2850]: Is it ok for having a lower cased L in there?

**Review by Nefertumm — APPROVED** [review]

**Comment by Nefertumm** [comment]
Also, tested and working!

**Comment by Nyeriah** [comment]
Conflicts

**Comment by FrancescoBorzi** [comment]
Conflicts solved.

Pro-tip: to make sure that conflicts like this have correctly been resolved, just double-check the diff of the file which had conflicts before and expect it to be unchanged: https://github.com/azerothcore/azerothcore-wotlk/pull/10301/files#diff-c73e471985fd90290ab0e85bf93c0884cb8f9586b11a14e28d0180fcec500ea8

**Comment by UltraNix** [comment]
@Kitzunu Done.

**Pull request merged**